src/uploadFile/pusher.ts

```diff
@@ -0,0 +1,168 @@
+import WebSocket from '../tools/sockets.node'
+
+import { FileInfo } from '../api/types'
+import { Status } from '../api/fromUrlStatus'
+import { Events } from './events'
+
+type AllStatuses =
+  | StatusErrorResponse
+  | StatusProgressResponse
+  | StatusSuccessResponse
+
+type StatusProgressResponse = {
+  status: Status.Progress
+  done: number
+  total: number
+}
+
+type StatusErrorResponse = {
+  status: Status.Error
+  msg: string
+  url: string
+}
+
+type StatusSuccessResponse = {
+  status: Status.Success
+} & FileInfo
+
+const response = (
+  type: 'progress' | 'success' | 'fail',
+  data: any
+): AllStatuses => {
+  if (type === 'success') {
+    return { status: Status.Success, ...data } as StatusSuccessResponse
+  }
+
+  if (type === 'progress') {
+    return { status: Status.Progress, ...data } as StatusProgressResponse
+  }
+
+  return { status: Status.Error, ...data } as StatusErrorResponse
+}
+
+const key = '79ae88bd931ea68464d9'
+const protocol =
+  typeof window !== 'undefined' && document.location.protocol !== 'https:'
+    ? 'ws:/'
+    : 'wss://'
+
+const port = protocol === 'wss://' ? 443 : 80
+
+const pusherUrl = `${protocol}ws.pusherapp.com:${port}/app/${key}?protocol=5&client=js&version=1.12.2`
```

Comment:
I think it's better to take this piece of code with url generation somewhere higher like here [./src/uploadFile/pushStrategy.ts](./src/uploadFile/pushStrategy.ts) and pass it at the pusher from it